File: ex05/ex05_07/ex05_07.cpp
#include <iostream>
using namespace std;

int main() {
  const double LOAN = 5400000;
  const double INTEREST_RATE = 0.812 / 100 /12;
  const int LOAN_TERM = 36; 
  const double INSTALMENT = LOAN / LOAN_TERM;
  double loan = LOAN; 
  double principal_plus_interest = 0.0;
  double interest = 0.0;
  double payment = 0.0;

  for (int i = 0; i < LOAN_TERM; i++) {
    interest = loan * INTEREST_RATE;
    payment = INSTALMENT + interest;
    loan -= INSTALMENT;
    principal_plus_interest += payment;
    cout << i << ": " << payment << " / " << interest << endl;
  }
  cout << "Principal plus interest: " << principal_plus_interest << endl;
  cout << "Total interest: " << principal_plus_interest - LOAN << endl;
  return 0;
}

File: ex05/ex05_07/ex05_07.py
def main():
  LOAN = 5400000
  INTEREST_RATE = 0.812 / 100 /12
  LOAN_TERM = 36 
  INSTALMENT = LOAN / LOAN_TERM
  loan = LOAN
  principal_plus_interest = 0.0
  interest = 0.0
  payment = 0.0

  for i in range(LOAN_TERM):
    interest = loan * INTEREST_RATE
    payment = INSTALMENT + interest
    loan -= INSTALMENT
    principal_plus_interest += payment
    print(str(i) + ": " + str(round(payment, 2)) + " / " + str(round(interest, 2)))
  
  print("Principal plus interest: " + str(round(principal_plus_interest)))
  print("Total interest: " + str((round(principal_plus_interest - LOAN, 2))))
  return 0

if __name__ == "__main__":
  main()